lib/mastermind/game.rb

```diff
@@ -0,0 +1,63 @@
+module Mastermind
+  class Game
+    class << self
+      include Instructions
+      include Gameplay
+      include Gamelog
+
+      def start
+        show_options_message
+        selected_option = gets.chomp
+
+        case selected_option
+        when 'p'
+          play_game
+        when 'i'
+          show_instruction
+        when 'l'
+          show_leader_board
+        when 'q'
+          quit_message
+          exit
+        else
+          invalid_entry
+        end
+      end
+
+      def play_game
+        select_difficulty_message
+        difficulty = gets.chomp
+        set_difficulty(difficulty)
+      end
+
+      def set_difficulty(difficulty)
+        case difficulty 
+        when 'b'
+          start_beginner_game ? start : nil
```

Comment:
Errr!! This is not right. Do all the recursion in `start_beginner_game`. Same goes for the next  lines.